jwst/extract_2d/grisms.py

```diff
@@ -111,6 +111,12 @@ def extract_tso_object(input_model,
     if len(available_orders) > 1:
         raise NotImplementedError("Multiple order extraction for TSO not currently implemented")
 
+    if input_model.meta.wcsinfo.siaf_xref_sci is None:
+        raise ValueError('XREF_SCI is missing.')
+
+    if input_model.meta.wcsinfo.siaf_yref_sci is None:
+        raise ValueError('YREF_SCI is missing.')
+
```

Comment:
... I was about to say change lines 132-133 to no longer subtract 1, because we're doing it earlier, but the subtraction that's done earlier is done on the values in the WCS transforms, not on the keyword values themselves, so the keyword values themselves will still be 1-indexed in lines 132-133 and hence still need to subtract 1.

One way around this would be to modify the meta data values themselves back in ``assign_wcs``, but I don't like th idea of modifying meta data values - they should always be kept at their original input values.

So is the change to subtract 1 in `assign_wcs/nircam.py` necessary after all?  @nden